.github/workflows/dashboard.yml

```diff
@@ -0,0 +1,71 @@
+name: Token Dashboard
+
+on:
+  pull_request:
+    branches: [ master ]
+    paths:
+    - solidity/dashboard/**
+
+# For pull_request_target:
+#  - Checkout with repository set to PR repo, ref same (see
+#    https://github.com/actions/checkout#usage).
+#  - Trigger from issue comment?
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.11.x]
+
+    if: startsWith(github.head_ref, 'releases/mainnet/token-dashboard/')
+    defaults:
+      run:
+        working-directory: solidity/dashboard
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
+    #- run: npm run lint
+    #- if: github.event_name == 'push' TODO uncomment when mainnet builds happen this way
+    #  run: npm run build --if-present
+    - if: github.event_name == 'pull_request'
+      run: npm run build
+      env:
+        PUBLIC_URL: /${{ github.head_ref }}
+    # A push event is a master merge; deploy to primary bucket.
+    # TODO uncomment when mainnet builds happen this way
+    #- if: github.event_name == 'push'
+    #  name: Deploy Master to GCP
+    #  uses: thesis/gcp-storage-bucket-action@v3.1.0
+    #  with:
+    #    service-key: ${{ secrets.KEEP_DASHBOARD_UPLOADER_SERVICE_KEY_JSON }}
+    #    project: keep-prd-210b
+    #    bucket-name: dashboard.keep.network
+    #    build-folder: solidity/dashboard/build
+    # A pull_request event is a PR; deploy to preview bucket.
+    - if: github.event_name == 'pull_request'
+      name: Deploy PR to GCP
```

Comment:
Can we mention in the step name that we're deploying it as a preview?

Reply:
Beyond the URL there's no real distinction, and the URL is handled in a separate step (the “post preview URL” one), so I'm tempted to keep it as-is. But if you feel strongly I can add it :)

Reply:
I'm afraid of confusion as the commented step above has a similar name `Deploy Master to GCP`, but only after looking at the `bucket-name` you can see the real difference between these two.

Reply:
```suggestion
      name: Deploy PR preview to GCP
```
How's that?

Reply:
Amazing!